pkg/asset/cluster/tfvars.go

```diff
@@ -36,24 +40,76 @@ func (t *TerraformVariables) Dependencies() []asset.Asset {
 		&installconfig.InstallConfig{},
 		&bootstrap.Bootstrap{},
 		&machine.Master{},
+		&machines.Master{},
 	}
 }
 
 // Generate generates the terraform.tfvars file.
 func (t *TerraformVariables) Generate(parents asset.Parents) error {
 	installConfig := &installconfig.InstallConfig{}
-	bootstrap := &bootstrap.Bootstrap{}
-	master := &machine.Master{}
-	parents.Get(installConfig, bootstrap, master)
+	bootstrapIgnAsset := &bootstrap.Bootstrap{}
+	masterIgnAsset := &machine.Master{}
+	mastersAsset := &machines.Master{}
+	parents.Get(installConfig, bootstrapIgnAsset, masterIgnAsset, mastersAsset)
 
-	bootstrapIgn := string(bootstrap.Files()[0].Data)
+	bootstrapIgn := string(bootstrapIgnAsset.Files()[0].Data)
 
-	masterIgn := string(master.Files()[0].Data)
+	masterIgn := string(masterIgnAsset.Files()[0].Data)
 
-	data, err := tfvars.TFVars(installConfig.Config, bootstrapIgn, masterIgn)
+	var mastersList metav1.List
+	err := yaml.Unmarshal(mastersAsset.MachinesRaw, &mastersList)
```

Comment:
can we change the master asset to be typed.. no need for this unamrshal? There were no consumers for that before, might be good time to add it...

Reply:
> can we change the master asset to be typed...

I'm fine with the `Unmarshal` here, especially while [we continue to generate the OpenStack machines via template][1].  But I can add a YAML unmarshal in `*Master.Generate` for the OpenStack machines if you'd rather push this up there and store structured information on the asset.

[1]: https://github.com/openshift/installer/blob/17ffd16d917ddbb7d1f0eeff9833289bff560437/pkg/asset/machines/master.go#L133

Reply:
The master asset is typed with 2b2e8c149.